idaes/apps/grid_integration/bidder.py

```diff
@@ -939,29 +968,34 @@ def _assemble_bids(self, model, power_var_name, energy_price_param_name, hour):
 
             if self.fixed_to_schedule:
                 bids[t][self.generator]["p_min"] = bids[t][self.generator]["p_max"]
-                bids[t][self.generator]["min_up_time"] = 0
-                bids[t][self.generator]["min_down_time"] = 0
                 bids[t][self.generator]["fixed_commitment"] = (
                     1 if bids[t][self.generator]["p_min"] > 0 else 0
                 )
-                bids[t][self.generator]["startup_fuel"] = [
-                    (bids[t][self.generator]["min_down_time"], 0)
-                ]
-                bids[t][self.generator]["startup_cost"] = [
-                    (bids[t][self.generator]["min_down_time"], 0)
-                ]
 
-            bids[t][self.generator]["p_cost"] = [
-                (bids[t][self.generator]["p_min"], 0),
-                (bids[t][self.generator]["p_max"], 0),
-            ]
+                if is_thermal:
```

Comment:
My suggestion is to add just a few comments here "When self-schedule for thermal generator, set startup times and costs to 0."